src/api.js

```diff
@@ -0,0 +1,797 @@
+@throw [String] SQLite Error
+ */
+Statement.prototype["step"] = function() {
+    var ret;
+    if (!this.stmt) {
+        throw "Statement closed";
+    }
+    this.pos = 1;
+    switch (ret = sqlite3_step(this.stmt)) {
+    case SQLITE_ROW:
+        return true;
+    case SQLITE_DONE:
+        return false;
+    default:
+        return this.db.handleError(ret);
+    }
+};
+
+Statement.prototype.getNumber = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_double(this.stmt, pos);
+};
+
+Statement.prototype.getString = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_text(this.stmt, pos);
+};
+
+Statement.prototype.getBlob = function(pos) {
+    var i, k, ptr, ref, result, size;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    size = sqlite3_column_bytes(this.stmt, pos);
+    ptr = sqlite3_column_blob(this.stmt, pos);
+    result = new Uint8Array(size);
+    for (i = k = 0, ref = size; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
+        result[i] = HEAP8[ptr + i];
+    }
+    return result;
+};
+
+/* Get one row of results of a statement.
+If the first parameter is not provided, step must have been called before get.
+@param [Array,Object] Optional: If set, the values will be bound to the statement, and it will be executed
+@return [Array<String,Number,Uint8Array,null>] One row of result
+
+@example Print all the rows of the table test to the console
+
+    var stmt = db.prepare("SELECT * FROM test");
+    while (stmt.step()) console.log(stmt.get());
+ */
+Statement.prototype["get"] = function(params) {
+    var field, k, ref, results1;
+    if (params != null) {
+        this["bind"](params) && this["step"]();
+    }
+    results1 = [];
+    for (field = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; field = 0 <= ref ? ++k : --k) {
+        switch (sqlite3_column_type(this.stmt, field)) {
+        case SQLITE_INTEGER:
+        case SQLITE_FLOAT:
+            results1.push(this.getNumber(field));
+            break;
+        case SQLITE_TEXT:
+            results1.push(this.getString(field));
+            break;
+        case SQLITE_BLOB:
+            results1.push(this.getBlob(field));
+            break;
+        default:
+            results1.push(null);
+        }
+    }
+    return results1;
+};
+
+/* Get the list of column names of a row of result of a statement.
+@return [Array<String>] The names of the columns
+@example
+
+    var stmt = db.prepare("SELECT 5 AS nbr, x'616200' AS data, NULL AS null_value;");
+    stmt.step(); // Execute the statement
+    console.log(stmt.getColumnNames()); // Will print ['nbr','data','null_value']
+ */
+Statement.prototype["getColumnNames"] = function() {
+    var i, k, ref, results1;
+    results1 = [];
+    for (i = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
```

Comment:
we can simply iterate in increasing order

Reply:
implemented suggestion in commit c79708ee4974f0a1cdf9cf78655b2a9b98ba974f